webcompat/api/uploads.py

```diff
@@ -86,6 +86,8 @@ def save(self):
         # Optimize further the image compression for these formats
         if self.image_object.format in ['JPEG', 'JPG', 'JPE', 'PNG']:
             save_parameters['optimize'] = True
+            # Convert PNG to JPEG. See issue #1051
+            file_dest = 'jpg'.join(file_dest.rsplit('png', 1))
```

Comment:
yes, due to pillow. Note that we could be more explicit. 
What it does here, is take the destination path, and change the `png` suffix by `jpg`. Because pillow is using the suffix for choosing the conversion format (at least it's one way it does it). If the file alreay has a `jpg`, it  doesn't change the string.